src/System.Runtime.Serialization.Formatters/tests/BinaryFormatterTests.cs

```diff
@@ -523,6 +523,8 @@ private static void SanityCheckBlob(object obj, string[] blobs)
             }
         }
 
+        private static int GetBlobIndex(int length) => PlatformDetection.IsFullFramework ? 1 : (length == 3 ? 2 : 0);
```

Comment:
Can you help me understand this?

Reply:
The test data for Serialization tests is (obj, string[])
We are referring to string[] as blobs array as it contains serialisations blobs.
So index 1 in that string array refers to the serialization blob of that type on Netfx.
Index 0 refers to netcoreapp2.0 or newer version if string blobs are just present for one of them
Index 2 if present contains the string blob for recent netcoreapp

Reply:
```
        private static int GetBlobIndexFromTargetFramework(int length, TargertFramework tf = TargertFramework.NetCoreApp)
        {
            if (PlatformDetection.IsFullFramework)
                tf = TargertFramework.Netfx;

            switch (tf)
            {
                case TargertFramework.Netfx :
                    return 0;
                case TargertFramework.NetCoreApp20 :
                    return 1;
                default: // Default platform is NetCoreApp
                    return length - 1;
            }
        }
```
```
    internal enum TargertFramework
    {
        Netfx = 0,
        NetCoreApp20 = 1,
        NetCoreApp = 2
    }
```
It `s scable as we can just add the index and also contains the helper function. we will need to swap some of the data in binaryTestData but i think its worth the effort. I can start swapping the blobs if you agree.

Reply:
I think such a enum could already exist somewhere in the test code in corefx. With _not scalable_ I meant your previous approach with the length == 3 comparison. What if we add another blob for Mono? What if we add different blobs for different versions of Mono? Would you then swap all the blobs again and add a custom logic for Mono? I don't think it's worth swapping them. You could just handle that in your code.

Reply:
But I'll leave it up to you, maybe you'll find a better way.